pkg/resources/usercluster/deployment.go

```diff
@@ -87,11 +85,16 @@ func DeploymentReconciler(data userclusterControllerData) reconciling.NamedDeplo
 	return func() (string, reconciling.DeploymentReconciler) {
 		return resources.UserClusterControllerDeploymentName, func(dep *appsv1.Deployment) (*appsv1.Deployment, error) {
 			dep.Name = resources.UserClusterControllerDeploymentName
-			dep.Labels = resources.BaseAppLabels(name, nil)
+			dep.Labels = resources.BaseAppLabels(resources.UserClusterControllerDeploymentName, nil)
 
 			dep.Spec.Replicas = resources.Int32(1)
```

Comment:
Your patch allows admins to configure the replica count, yet you're not using the one that is configured. Is that an oversight?

Reply:
https://github.com/kubermatic/kubermatic/pull/12211/files#diff-fc8f73fda439bfb46949c7196ce280cfe5393c27b629249b8d342a5ff4f981f3R92 it's actually done just two lines below, but only if it the override is actually set.